modules/cas_cache/layer_cache_management.c

```diff
@@ -2353,12 +2353,27 @@ int cache_mngt_activate(struct ocf_mngt_cache_standby_activate_config *cfg,
 	if (result == -KCAS_ERR_WAITING_INTERRUPTED)
 		goto out_cache_put;
 
-	cas_lazy_thread_stop(context->rollback_thread);
+	result = _cache_start_finalize(cache, -1, true);
+	if (result)
+        goto err;
 
+    cas_lazy_thread_stop(context->rollback_thread);
 	if (result)
```

Comment:
If the activation and start_finalize have been successful you don't want to stop the cache you just activated so I guess the condition should be `result == 0`